src/providers/pass-certificate-validator/pass-certificate-validator.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class PassCertificateValidatorProvider {
+
+  isPassCertificateValid(certificate: string | null): boolean {
+    if (!certificate || certificate.length !== 8) {
+      return false;
+    }
+    if (!this.isLetter(certificate[0])) {
+      return false;
+    }
+
+    const checkDigit = this.calculateMod37CheckDigit(certificate);
+
+    if (checkDigit === 'invalid') {
+      return false;
+    }
+
+        // compare entered checkdigit with generated one
+    if (checkDigit !== certificate.toUpperCase()[7]) {
+      return false;
+    }
+
+    return true;
+  }
+
+  isLetter(char: string): boolean {
+    return char.length === 1 && char.match(/[a-z]/i) !== null;
+  }
+
+  calculateMod37CheckDigit(certificate: string): string {
```

Comment:
Would be cool if you added a description for this function. Just a few words in a comment about the logic, so that it's more easily understandable.

Reply:
added description